flocker/node/agents/blockdevice.py

```diff
@@ -852,7 +875,7 @@ def run(self, deployer):
         check_output(["mkfs", "-t", "ext4", device.path])
 
         # This duplicates MountBlockDevice now.
-        self.mountpoint.makedirs()
+        _create_mount_directory(self.mountpoint)
```

Comment:
Does this work? 
I thought you had to set the permissions on the mounted filesystem. 
ie the directory permissions of the mount root need to be stored in the mounted filesystem.